### PR TITLE
PR-12 Extend Webrest API to include PatientQueue Object

### PR DESCRIPTION
**Pull request by slubwama**
https://issues.openmrs.org/browse/PQ-12

### CODE REVIEW

**Comment by slubwama** [comment]
@dkayiwa  help review

File: omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.util.OpenmrsUtil;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+
+@Resource(name = RestConstants.VERSION_1 + "/PatientQueue", supportedClass = PatientQueue.class, supportedOpenmrsVersions = {
+        "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*", "2.2.*", "2.3.*", "2.4.*" })
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java]: Don't you want this to run on 2.5.0?

Comment (dkayiwa) [reply omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java]: For `/PatientQueue` do you mind following our convention of naming webservices resources?

**Comment by dkayiwa** [comment]
Do you have a use case for this?

File: omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java (hunk 1 above)
Comment (slubwama) [reply omod/src/main/java/org/openmrs/module/patientqueueing/web/resource/PatientQueueResource.java]: @dkayiwa done responding to comments.

**Comment by dkayiwa** [comment]
@slubwama do you mind adding some controller tests like the ones here? https://github.com/openmrs/openmrs-module-webservices.rest/tree/master/omod-1.9/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9

**Comment by slubwama** [comment]
@dkayiwa  I am trying to include tests but I am having trouble with dependencies kindly advise. 
https://pastebin.com/EGFas6Rp

**Comment by dkayiwa** [comment]
@slubwama i have made a couple of commits to fix the dependency errors. You can now resume working on the two tests that you included in this pull request. I ignored them just for the sake of keeping the build green. Un ignore them after you have finished them.

**Pull request merged**